### PR TITLE
Clarify which assembly sets up compatibility modes

### LINKED ISSUE #5075: Runtime changes with assemblies compiled against mixed Framework versions
In the *Runtime changes* section, I read that if:
* the entry assembly is compiled for .NET Framework 4.0
* the installed runtime is .NET Framework 4.5

then the code will run in .NET Framework 4.0 *compatibility mode*.

Please clarify what would happen if the entry assembly loads an assembly compiled for .NET Framework 4.5.
Would it run in compatibility mode too?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4ee6a407-16fe-6537-e587-99908c9a8248
* Version Independent ID: 920fcce6-df42-ff50-dae2-cd7110a8a2f3
* Content: [Application Compatibility in the .NET Framework](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/application-compatibility#runtime-changes)
* Content Source: [docs/framework/migration-guide/application-compatibility.md](https://github.com/dotnet/docs/blob/master/docs/framework/migration-guide/application-compatibility.md)
* Service: **unspecified**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### PR DESCRIPTION
**Pull request by twsouthwick**
Fixes #5075

### CODE REVIEW

File: docs/framework/migration-guide/application-compatibility.md [hunk 1]
```diff
@@ -33,7 +33,8 @@ When running on a newer version than what was targeted, the .NET Framework will
 use quirked behavior to mimic the older targeted version. In other words, the
 app will run on the newer version of the Framework, but act as if it's running
 on the older version. Many of the compatibility issues between versions of the .NET
-Framework are mitigated through this quirking model.
+Framework are mitigated through this quirking model. This compatibility mode is determined
+by the entry assembly for the AppDomain that the code is running in.
```
Comment (svick) [inline docs/framework/migration-guide/application-compatibility.md]: Would it make sense to add that the assembly in question is usually the executable for the application?

File: docs/framework/migration-guide/application-compatibility.md [hunk 2]
```diff
@@ -33,7 +33,10 @@ When running on a newer version than what was targeted, the .NET Framework will
 use quirked behavior to mimic the older targeted version. In other words, the
 app will run on the newer version of the Framework, but act as if it's running
 on the older version. Many of the compatibility issues between versions of the .NET
-Framework are mitigated through this quirking model.
+Framework are mitigated through this quirking model. This compatibility mode is determined
+by the entry assembly for the AppDomain that the code is running in. For example, in the case
+of an executable, the framework the executable was targeted against is the compatibility
```
Comment (rpetrusha) [inline docs/framework/migration-guide/application-compatibility.md]: was targeted against --> targets

File: docs/framework/migration-guide/application-compatibility.md [hunk 3]
```diff
@@ -33,7 +33,10 @@ When running on a newer version than what was targeted, the .NET Framework will
...
+mode all assemblies in that AppDomain will run under.
```
Comment (rpetrusha) [inline docs/framework/migration-guide/application-compatibility.md]: nit: compatibility mode is a general term that doesn't specify a version. I think I'd rephrase (starting from the current "This compatibility mode..."):

The version of the .NET Framework that an application targets is determined by the target version of the entry assembly for the application domain that the code is running in. All additional assemblies loaded in that application domain target that .NET Framework version.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks, @twsouthwick; this is a really useful clarification. I left two suggested changes.

File: docs/framework/migration-guide/application-compatibility.md (hunk 2 above)
Comment (twsouthwick) [reply docs/framework/migration-guide/application-compatibility.md]: fixed

**Review by rpetrusha — APPROVED** [review]
Thanks for making these additional changes, @twsouthwick. This looks really good. I'll merge your PR now.
